src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -184,6 +185,17 @@ const OrderRow: React.FC<Props> = props => {
   )
   const expiresOn = order.validUntil.toLocaleString() // TODO: make it nice like, 3 min, 1 day, etc
 
+  useEffect(() => {
```

Comment:
Shouldn't `overBalances` (set) be what the hook returns? This way you can use it in the rows, and also to display the legend if it's not an empty set.

Why we need the update logic? Maybe is just to use the orders in the hook constructor param?

Reply:
> Shouldn't overBalances (set) be what the hook returns? This way you can use it in the rows, and also to display the legend if it's not an empty set.

I don't think so, because I don't need to use the Set contents anywhere here. It's only used to track whether there's an order over balance.

> Why we need the update logic? Maybe is just to use the orders in the hook constructor param?

Because the `overBalance` calculation is done on `OrderRow`, while `OrderWidget` is the one holding the state and displaying the low balance indicator.